llvm/include/llvm/ADT/SetOperations.h

```diff
@@ -60,6 +60,13 @@ template <class S1Ty, class S2Ty> void set_intersect(S1Ty &S1, const S2Ty &S2) {
   auto Pred = [&S2](const auto &E) { return !S2.count(E); };
   if constexpr (detail::HasMemberRemoveIf<S1Ty, decltype(Pred)>) {
     S1.remove_if(Pred);
+  } else if constexpr (detail::HasMemberEraseIter<S1Ty>) {
+    typename S1Ty::iterator Next;
+    for (typename S1Ty::iterator I = S1.begin(); I != S1.end(); I = Next) {
+      Next = std::next(I);
```

Comment:
nit: Just below we call `++I`. Do we need the `std::next` then?

Reply:
We can use `std::next` there, too.  Are you suggesting replacing `++I` with `std::next` for consistency?

Reply:
I'd use `++I` for consistency if it's known to work in this context, otherwise use `next` in both places

Reply:
Thank you for the clarification.  Here, I need to increment `I`, store the result in a another variable, and then call `S1.erase(I)` with the original value of `I`.  I'll use `std::next` in both places then.

Reply:
Wait, @nikic suggested to drop the "erase-by-value" code path, so there is no "both places" anymore.